File: Lehrverwaltung/server/src/main/java/tie/lehrlinge/lehrverwaltung/server/model/ExerciseModel.java
package tie.lehrlinge.lehrverwaltung.server.model;

import java.util.List;

import lombok.Getter;
import lombok.NonNull;
import lombok.RequiredArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@RequiredArgsConstructor
public class ExerciseModel {
  private  @NonNull int id;

  private @NonNull String title;
  private @NonNull String language;
  private @NonNull String requirement;
  private String description;
  private @NonNull List<String> links;
  private byte[] textFile;
  private String name;
  private byte[] data;
  private String type;
}

File: Lehrverwaltung/client/src/app/core/models/exercise/ExerciseModel.ts
export class ExerciseModel {

  id: number;
  title: string;
  language: string;
  requirement: string;
  description: string;
  links: string[];
  textFile: string;
  name: string;
  type: string;
  data: string;

  constructor(id: number, title: string, language: string, requirement: string, description: string, links: string[], textFile: string, name: string, type: string, data: string) {
    this.id = id;
    this.title = title;
    this.language = language;
    this.requirement = requirement;
    this.description = description;
    this.links = links;
    this.textFile = textFile;
    this.name = name;
    this.type = type;
    this.data = data;
  }
}
